src/main/java/frc/robot/subsystems/drive/states/LinearDriveState.java

```diff
@@ -0,0 +1,266 @@
+                JsonConstants.redFieldLocations.redReefOTF9Rotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueReefOTF9Translation,
+                JsonConstants.blueFieldLocations.blueReefOTF9Rotation);
+      case Reef10:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redReefOTF10Translation,
+                JsonConstants.redFieldLocations.redReefOTF10Rotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueReefOTF10Translation,
+                JsonConstants.blueFieldLocations.blueReefOTF10Rotation);
+      case Reef11:
+      case Algae5:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redReefOTF11Translation,
+                JsonConstants.redFieldLocations.redReefOTF11Rotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueReefOTF11Translation,
+                JsonConstants.blueFieldLocations.blueReefOTF11Rotation);
+      case CoralStationRight:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redCoralStationRightTranslation,
+                JsonConstants.redFieldLocations.redCoralStationRightRotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueCoralStationRightTranslation,
+                JsonConstants.blueFieldLocations.blueCoralStationRightRotation);
+      case CoralStationLeft:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redCoralStationLeftTranslation,
+                JsonConstants.redFieldLocations.redCoralStationLeftRotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueCoralStationLeftTranslation,
+                JsonConstants.blueFieldLocations.blueCoralStationLeftRotation);
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  public void periodic() {
+    Pose2d currentPose = drive.getPose();
+
+    double currentDistance = currentPose.getTranslation().getDistance(goalPose.getTranslation());
+
+    if (withinRange(currentPose, goalPose, 0.05)) {
+      drive.setGoalSpeeds(new ChassisSpeeds(), true);
+
+      if (drive.isDesiredLocationReef()) {
+        drive.fireTrigger(DriveTrigger.BeginLineup);
+      } else {
+        drive.fireTrigger(DriveTrigger.CancelLinear);
+      }
+      return;
+    }
+
+    // DRIVE
+    double ffScaler =
+        MathUtil.clamp(
+            (currentDistance - m_ffMinRadius) / (m_ffMaxRadius - m_ffMinRadius), 0.0, 1.0);
+    double driveVelocityScalar =
+        driveController.getSetpoint().velocity * ffScaler
+            + driveController.calculate(currentDistance, 0.0);
+    if (currentDistance < driveController.getPositionTolerance()) {
+      driveVelocityScalar = 0.0;
+    }
```

Comment:
I'm not sure why this logic is required? Can we pass in the distance error and current velocity to the profiled pid and let it handle the rest?